src/main/java/com/okta/sdk/framework/ApiClient.java

```diff
@@ -45,18 +51,19 @@
     private AtomicReference<HttpResponse> lastHttpResponseRef = new AtomicReference<HttpResponse>();
 
     ////////////////////////////////////////////
-    // CONSTRUCTORS
+    // CONSTRUCTOR
     ////////////////////////////////////////////
 
     public ApiClient(ApiClientConfiguration config) {
-        this(HttpClientBuilder.create()
-                        .setUserAgent("OktaSDKJava_v" + Utils.getSdkVersion())
-                        .disableCookieManagement().build(),
-                config);
-    }
+        PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+        connectionManager.closeExpiredConnections();
+        connectionManager.closeIdleConnections(TTL, TimeUnit.SECONDS);
```

Comment:
When you say periodically, what should be that period? What if the server closes the connection in that period but the client still uses that connection? The request will then fail. Retrying the requests seems like the safest approach to serve requests.  
Also, I tried the setConnectionTimeToLive and it did not solve the issue.